lib/internal/Magento/Framework/EntityManager/Operation/ExtensionPool.php

```diff
@@ -57,4 +57,27 @@ public function getActions($entityType, $actionName)
         }
         return $actions;
     }
+
+    /**
+     * Add action to extension pool
+     *
+     * $action array should be in the format:
+     * [
+     *     'read' => [
+     *         'scopeDataReader' => 'Magento\Catalog\Model\ResourceModel\Category\ScopeData\ReadHandler'
+     *     ]
+     * ]
+     *
+     * @param $entityType
+     * @param array $action
+     * @return $this
+     */
+    public function addAction($entityType, array $action)
```

Comment:
Currently actions can only be added via a `di.xml` file. This allows actions to be added programatically.